lib/helper.c

```diff
@@ -875,9 +877,11 @@ ipmi_start_daemon(struct ipmi_intf *intf)
 	}
 
 	fd = open("/dev/null", O_RDWR);
-	assert(0 == fd);
-	dup(fd);
-	dup(fd);
+	assert(STDIN_FILENO == fd);
+	ret = dup(fd);
+	assert(STDOUT_FILENO == ret);
+	ret = dup(fd);
+	assert(STDERR_FILENO == ret);
```

Comment:
Same here. Don't use assertions. Check return values of `dup()` and print out a errno-based message on errors.

Reply:
ok